server/opts.go

```diff
@@ -1680,6 +1680,53 @@ func parseJetStreamForAccount(v interface{}, acc *Account, errors *[]error, warn
 	return nil
 }
 
+func getInterfaceValue(v interface{}) (int64, error) {
+	_, ok := v.(int64)
+	if ok {
+		return v.(int64), nil
+	}
+
+	_, ok = v.(string)
+	if !ok {
+		return 0, fmt.Errorf("must be int64 or string")
+	}
+
+	num, err := stringStorageSize(v.(string))
+	if err != nil {
+		return 0, err
+	}
+
+	return int64(num), nil
+}
+
+func stringStorageSize(s string) (int, error) {
+	suffix := strings.ToUpper(s[len(s)-1:])
+	prefix := s[:len(s)-1]
+	var num int
+	var err error
+	switch suffix {
+	case "K":
+		num, err = strconv.Atoi(prefix)
+		num = num * 1024
+	case "M":
+		num, err = strconv.Atoi(prefix)
+		num = num * 1048576
```

Comment:
I would have done `1024*1024` (same for others). The compiler will do the operation once, but it is easier to read for a dev.

Reply:
yeah, I wasn't sure which was better and guessed wrong 😛 I"ll change it to the other way.